orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/poller/SaltBootstrap.java

```diff
@@ -48,16 +48,22 @@ public class SaltBootstrap implements OrchestratorBootstrap {
 
     private final BootstrapParams params;
 
+    private final SaltStates saltStates;
+
     private Set<Node> targets;
 
-    public SaltBootstrap(SaltConnector sc, Collection<SaltConnector> saltConnectors, List<GatewayConfig> allGatewayConfigs, Set<Node> targets,
+    /*
+        Intentionally Package-private to be able to verify construction later
```

Comment:
could you elaborate?

Reply:
PowerMockito checked for the creation of the `SaltBootsrap class`
In regular Mockito, verifying of constructor call is not possible so to workaround this I introduced a Factory class that instantiates `SaltBootstrap` and now the factory method can be verified.